app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,51 @@
+class AddressesController < ApplicationController
+    def index 
+        address = Address.all 
+        render json: address 
+    end
+
+    def show
+      id = params[:id]
+      address = Address.find(id)
+      render json: address
+    end
+
+    def new 
+        address = Address.new 
+        render json: address
+    end
+  
+    def create
+      # param keys may subject to change depending on the body of the post request
+      address = Address.create(address_params)
```

Comment:
Should replace ".create" with ".new". Using the "create" method will immediately invoke all of the validations methods in the Address model and may not actually make a new address and save it.